doc/proposal/thrift-parameter-validation-proposal.md

```diff
@@ -0,0 +1,180 @@
+| eq           | equals to (`==`)                 | i8, i16, i32, i64, f64, string, bool | same with value        | -                  |
+| ne           | not equals to (`!=`)             | i8, i16, i32, i64, f64, string, bool | same with value        | -                  |
+| lt           | less than (`<`)                  | i8, i16, i32, i64, f64               | same with value        | -                  |
+| le           | less equal (`<=`)                | i8, i16, i32, i64, f64               | same with value        | -                  |
+| gt           | greater than (`>`)               | i8, i16, i32, i64, f64               | same with value        | -                  |
+| ge           | greater equal (`>=`)             | i8, i16, i32, i64, f64               | same with value        | -                  |
+| in           | within given container           | i8, i16, i32, i64, f64               | same with value        | -                  |
+| not_in       | not within given container       | i8, i16, i32, i64, f64               | same with value        | -                  |
+| elem         | field's element constraint       | any                                  | list, set              | support            |
+| key          | field's element key constraint   | any                                  | map                    | support            |
+| value        | field's element value constraint | any                                  | map                    | support            |
+| min_size     | minimal length                   | i8, i16, i32, i64                    | string, list, set, map | -                  |
+| max_size     | maximal length                   | i8, i16, i32, i64                    | string, list, set, map | -                  |
+| prefix       | field prefix must be             | string                               | string                 | -                  |
+| suffix       | suffix must be                   | string                               | string                 | -                  |
+| contain      | must contain                     | string                               | string                 | -                  |
+| not_contain  | must not contain                 | string                               | string                 | -                  |
+
+- Secodary validator is a successive validator, usually used at container-type field. See below Set/List/Map examples.
+
+#### 3.2.2 IDL example
+- Number
+```
+struct NumericDemo{
+    1: double Value (validator.gte = "1000.1", validator.lte = "10000.1")
+    2: i8 Type (validator.in = "[1, 2, 4]")
+}
+```
+- String/Binary
+``` 
+struct StringDemo{
+    1: string Uninitialized (vt.const = "abc")
+    2: string Name (vt.min_size = "6", vt.max_size = "12")
+    3: string SomeStuffs (vt.pattern = "[0-9A-Za-z]+")
```

Comment:
I don't see `pattern` defined in the table. for a regexp there are so many flavors, I'd strongly suggest to either define the flavor here, or just remove it altogether.

Reply:
Considering that it will be implemented in many languages, it is more appropriate to use BRE (basic regular expression) here. I will add it to the table.